src/main/java/org/carlmontrobotics/robotcode2023/commands/ArmPeriodic.java

```diff
@@ -29,47 +29,67 @@ public ArmPeriodic(Arm armSubsystem, DoubleSupplier arm, DoubleSupplier wrist) {
 
   // Called when the command is initially scheduled.
   @Override
-  public void initialize() {}
+  public void initialize() {
+  }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
     double[] speeds = getRequestedSpeeds();
-    armSubsystem.driveArm(speeds[0], 0);
-    armSubsystem.driveWrist(speeds[1], 0);
-    armSubsystem.setArmTarget(armSubsystem.getArmPos());
-    armSubsystem.setWristTarget(armSubsystem.getWristPos());
+    // if driven by joysticks, run only feedforward
+    if (speeds[0] == 0) 
+      armSubsystem.driveArm(0, 0, true);
+    else {
+      armSubsystem.driveArm(speeds[ARM], 0, false);
+      armSubsystem.setArmTarget(armSubsystem.getArmPos());
+    }
+    if (speeds[1] == 0)
+      armSubsystem.driveWrist(0, 0, true);
+    else {
+      armSubsystem.driveWrist(speeds[WRIST], 0, false);
+      armSubsystem.setWristTarget(armSubsystem.getWristPos());
+    }
```

Comment:
I recommend that you replace all of this with something that just updates the target positions based on the current positions, the requested speeds, and the timestep. Then let the `Arm` subsystem worry about actually getting to the new desired position.

Reply:
Even better, pass the Arm subsystem a goal which is a `TrapezoidalProfile.State` containing the desired position and the requested speed and have the Arm subsystem execute a motion profile to achieve that state.